libraries/AP_Baro/AP_Baro_atmosphere.cpp

```diff
@@ -0,0 +1,388 @@
+#include "AP_Baro.h"
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  atmospheric model options
+ */
+
+#ifndef HAL_BARO_ATMOSPHERIC_TABLE
+// default to using tables when doing double precision EKF (which implies more flash space and faster MCU)
+// this allows for non-table testing with --ekf-single configure option
+#define HAL_BARO_ATMOSPHERIC_TABLE HAL_WITH_EKF_DOUBLE
+#endif
+
+/*
+  return altitude difference in meters between current pressure and a
+  given base_pressure in Pascal. This is a simple atmospheric model
+  good to about 11km AMSL. Enable the table based function for higher altitudes
+*/
+float AP_Baro::get_altitude_difference_function(float base_pressure, float pressure) const
+{
+    float ret;
+    float temp    = get_ground_temperature() + C_TO_KELVIN;
+    float scaling = pressure / base_pressure;
+
+    // This is an exact calculation that is within +-2.5m of the standard
+    // atmosphere tables in the troposphere (up to 11,000 m amsl).
+    ret = 153.8462f * temp * (1.0f - expf(0.190259f * logf(scaling)));
+
+    return ret;
+}
+
+#if HAL_BARO_ATMOSPHERIC_TABLE || CONFIG_HAL_BOARD == HAL_BOARD_SITL
+/*
+  MIL-STD 3013 (1962 U.S. Standard Atmosphere) in Tabular Form.
```

Comment:
Why not the International Standard Atmosphere as adopted by ICAO?

Reply:
i've used the one recommended for the high altitude testing as recommended by a aerodynamics specialist

Reply:
A stated goal was to give the same reading as calibrated sensors in manned aircraft give.  I believe they're calibrated against that ICAO standard.

Reply:
The ICAO according to wiki (famous last words) says it uses the USA 1976 model up to 85km which is what I used for my model of this. 

Unfortunately, I cant find a copy of the ICAO that doesnt cost $$ to compare constants and such.